mobilecoind/src/service.rs

```diff
@@ -520,6 +520,47 @@ impl<T: BlockchainConnection + UserTxConnection + 'static> ServiceApi<T> {
         Ok(response)
     }
 
+    fn read_address_code_impl(
+        &mut self,
+        request: mc_mobilecoind_api::ReadAddressCodeRequest,
+    ) -> Result<mc_mobilecoind_api::ReadAddressCodeResponse, RpcStatus> {
+        let request_wrapper = mc_mobilecoind_api::printable::PrintableWrapper::b58_decode(
+            request.get_b58_code().to_string(),
+        )
+        .map_err(|err| rpc_internal_error("PrintableWrapper_b58_decode", err, &self.logger))?;
+
+        if request_wrapper.has_public_address() {
+            let public_address = request_wrapper.get_public_address();
+            let mut response = mc_mobilecoind_api::ReadAddressCodeResponse::new();
+            response.set_receiver(public_address.clone());
+            Ok(response)
+        } else {
```

Comment:
I believe this should also support `has_payment_request` similarly to https://github.com/mobilecoinofficial/mobilecoin/blob/39aeb5a3f4d4ec89a08be973b20cca2b86380605/mobilecoind/src/service.rs#L364

If someone receives a b58 string they might not know if its a address code or payment request code, but both have a payable public address and they should be able to get it in both cases.